cypress/support/components/ScorerModal.js

```diff
@@ -18,14 +18,22 @@ export function ClickAction(expectedResponse)
   VerifyLastChatMessage(expectedResponse)
 }
 
+// Currently this function is known to work ONLY for:
+// TEXT and END_SESSION Actions.
 export function VerifyLastChatMessage(expectedMessage)
 {
   var expectedUtterance = expectedMessage.replace(/'/g, "’")
-
-  cy.Get('[data-testid="web-chat-utterances"]').then(elements => {
-    cy.wrap(elements[elements.length - 1]).within(e => {
-      cy.get('div.format-markdown > p').should('have.text', expectedUtterance)
-    })})
+  cy.Get('[data-testid="web-chat-utterances"]').then(elements => 
+  {
+    var element = elements[elements.length - 1]
+    if (!Cypress.$(element).find('div.wc-adaptive-card').length) {
+      expect(helpers.RemoveMarkup(Cypress.$(element).find('p').text())).to.equal(expectedUtterance)
+    } else {
+      if (expectedUtterance && (Cypress.$(element).html().indexOf(helpers.RemoveMarkup(expectedUtterance)) < 0)) {
```

Comment:
This is backwards...You need to remove the markup from the UI Element NOT from the "expectedUtterance" that was basically passed into this method.

Reply:
Actually lines 29, 31 & 32 can go away and it should work for all cases.